### PR TITLE
[13.0][ENH] l10n_th_withholding_tax_report : generate text file

### PR DESCRIPTION
**Pull request by Saran440**
- Change generate csv file to text file by core odoo
- Add test script
- Text file is not complete (Don't have format)

### CODE REVIEW

File: l10n_th_withholding_tax_report/README.rst [hunk 1]
```diff
@@ -45,20 +45,21 @@ To use this module, you need to:
 * View - Report type html.
   Export PDF - Report type pdf.
   Export XLSX - Report type xlsx.
+  Export TXT - Report type text.
 
 Changelog
```
Comment (kittiu) [inline l10n_th_withholding_tax_report/README.rst:50]: Thanks for seeing that the history log is not updated.
But normally, we want to keep the v12 history log too. So just add the new v13 log.

File: l10n_th_withholding_tax_report/report/report_withholding_tax_csv.py [hunk 2]
```diff
@@ -14,53 +13,6 @@
     _logger.debug("Can not import csvwriter`.")
 
 
-class WithholdingTaxCSV(models.AbstractModel):
```
Comment (kittiu) [inline l10n_th_withholding_tax_report/report/report_withholding_tax_csv.py:17]: Can you explain whey this is being removed?

Comment (Saran440) [reply l10n_th_withholding_tax_report/report/report_withholding_tax_csv.py:17]: Because this models is not used. I don't know why to create this models?

**Review by kittiu — APPROVED** [review]

**Comment by ps-tubtim** [comment]
@Saran440 Could you add pnd1, pnd3a in `income_tax_form` and `DICTPND` like l10n_th_withholding_tax_cert module? https://github.com/OCA/l10n-thailand/blob/b7de65346d67041a66e73a408b620a5955d8ee37/l10n_th_withholding_tax_cert/models/withholding_tax_cert.py#L8

**Review by ps-tubtim — APPROVED** [review]

**Comment by Saran440** [comment]
@kittiu Could you review and merge it? this PR add test script and change csv file to text file.

File: l10n_th_withholding_tax_report/data/report_data.xml [hunk 3]
```diff
@@ -18,14 +18,15 @@
...
         model="withholding.tax.report"
-        string="Withholding Tax Report (CSV)"
-        report_type="csv"
-        name="report_withholding_tax_csv"
-        file="report_withholding_tax_csv"
+        string="Withholding Tax Report (TXT)"
+        report_type="qweb-text"
+        name="l10n_th_withholding_tax_report.report_withholding_tax_text"
+        file="l10n_th_withholding_tax_report.report_withholding_tax_text"
+        print_report_name="'%sOZZZZZ%s%s0000' % (object.company_id.vat, object.format_pnd(object.income_tax_form), object.format_date())"
```
Comment (kittiu) [inline l10n_th_withholding_tax_report/data/report_data.xml]: Change to more readable name. I..e, WHT-xxx-yyy-zzz (I don't know, you can suggest).

**Comment by Saran440** [comment]
@kittiu I fix filename in pdf and text file to pattern `WHT-<type>-<date from>` i.e. WHT-P03-256307
Thank you for your review.

**Summary**
PDF > WHT-P03-256307.pdf
TEXT > WHT-P03-256307.txt
XLSX > withholding_tax.xlsx

**Pull request merged**